### PR TITLE
Update doc for FinalClass:#7609

### PR DESCRIPTION
**Pull request by ek08**
Issue:#7609

### CODE REVIEW

**Comment by romani** [comment]
Please restore original formatting, and keep changed only lines you have to update.

Please review CI failures, and resolve them.
Run `mvn verify` on local it will guide you what to fix.

**Comment by ek08** [comment]
Verify

mvn verify gives a lot of errors

File: src/xdocs/config_design.xml [hunk 1]
```diff
@@ -289,39 +289,43 @@
...
-                    public MyClass(String str){
-                    this.str=str;
-                    a=0;
-                    } //OK,since it has a public constructor
-                    }
+                    <source>
+                        final class MyClass{
+                        private MyClass(){ } //OK
+                        }
+                        class MyClass(){
```
Comment (AkMo3) [inline src/xdocs/config_design.xml:296]: Please remove the parenthesis in the class. It is causing the file not to build.

**Review by AkMo3 — COMMENTED** [review]
Also you need to update the javadocs for this change. It would also be great if you can change the format of code to make it easy to read. @romani Please correct me if I am wrong.

**Comment by AkMo3** [comment]
Also change the commit message, it would also cause for mvn verify to fail. Squash into a single message with proper format which is usually Issue #{ISSUE_NO}: {COMMIT_DISCREPTION}

**Comment by ek08** [comment]
I run mvn verify -X
 Failures: 
[ERROR]   CommitValidationTest.testCommitMessageHasProperStructure:148 Commit 768b33bf448d16b81e66c306ee0c597255fd4367 message: "Issue:#7609:Update doc for FinalClass\n" is invalid
Proper commit message should adhere to the following rules:
    1) Must match one of the following patterns:
        ^Issue #\d+: .*$
        ^Pull #\d+: .*$
        ^(minor|config|infra|doc|spelling|dependency): .*$
    2) It contains only one line of text
    3) Must not end with a period, space, or tab
    4) Commit message should be less than or equal to 200 characters

The rule broken was: 1
[ERROR] Errors: 
[ERROR]   XdocsJavaDocsTest.testAllCheckSectionJavaDocs:182->examineCheckSection:209 »  ...
[INFO] 
[ERROR] Tests run: 3406, Failures: 1, Errors: 1, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  02:18 min
[INFO] Finished at: 2020-11-18T19:46:22+05:30
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.2:test (default-test) on project checkstyle: There are test failures.
[ERROR] 
[ERROR] Please refer to /home/ehtesham/IdeaProjects/checkstyle/target/surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.2:test (default-test) on project checkstyle: There are test failures.

Please refer to /home/ehtesham/IdeaProjects/checkstyle/target/surefire-reports for the individual test results.
Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.apache.maven.plugin.MojoFailureException: There are test failures.

Please refer to /home/ehtesham/IdeaProjects/checkstyle/target/surefire-reports for the individual test results.
Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
    at org.apache.maven.plugin.surefire.SurefireHelper.throwException (SurefireHelper.java:289)
    at org.apache.maven.plugin.surefire.SurefireHelper.reportExecution (SurefireHelper.java:161)
    at org.apache.maven.plugin.surefire.SurefirePlugin.handleSummary (SurefirePlugin.java:364)
    at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked (AbstractSurefireMojo.java:1041)
    at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute (AbstractSurefireMojo.java:857)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)

This is what I get as output .One I understood is due to the wrong commit message but I can't get the other one.

**Comment by romani** [comment]
open IDE, open class XdocsJavaDocsTest , execute testAllCheckSectionJavaDocs out put will be more understandable.

**Comment by ek08** [comment]
> Also you need to update the javadocs for this change. It would also be great if you can change the format of code to make it easy to read. @romani Please correct me if I am wrong.

Can u please help me to understand which file I need to change in the javadocs. I have just changed the config_design.xml in the xdocs

**Comment by romani** [comment]
https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/design/FinalClassCheck.java

**Comment by AkMo3** [comment]
> 
> 
> I run mvn verify -X
> Failures:
> [ERROR] CommitValidationTest.testCommitMessageHasProperStructure:148 Commit 768b33bf448d16b81e66c306ee0c597255fd4367 message: "Issue:#7609:Update doc for FinalClass\n" is invalid
> Proper commit message should adhere to the following rules:
> 1) Must match one of the following patterns:
> ^Issue #\d+: ._$ ^Pull #\d+: ._$
> ^(minor|config|infra|doc|spelling|dependency): .*$
> 2) It contains only one line of text
> 3) Must not end with a period, space, or tab
> 4) Commit message should be less than or equal to 200 characters
> 
> The rule broken was: 1
> [ERROR] Errors:
> [ERROR] XdocsJavaDocsTest.testAllCheckSectionJavaDocs:182->examineCheckSection:209 » ...
> [INFO]
> [ERROR] Tests run: 3406, Failures: 1, Errors: 1, Skipped: 0
> [INFO]
> [INFO] ------------------------------------------------------------------------
> [INFO] BUILD FAILURE
> [INFO] ------------------------------------------------------------------------
> [INFO] Total time: 02:18 min
> [INFO] Finished at: 2020-11-18T19:46:22+05:30
> [INFO] ------------------------------------------------------------------------
> [ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.2:test (default-test) on project checkstyle: There are test failures.
> [ERROR]
> [ERROR] Please refer to /home/ehtesham/IdeaProjects/checkstyle/target/surefire-reports for the individual test results.
> [ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
> [ERROR] -> [Help 1]
> org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.2:test (default-test) on project checkstyle: There are test failures.
> 
> Please refer to /home/ehtesham/IdeaProjects/checkstyle/target/surefire-reports for the individual test results.
> Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
> at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
> at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
> at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
> at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
> at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
> at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
> at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
> at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
> at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
> at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
> at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
> at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
> at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
> at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
> at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
> at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
> at java.lang.reflect.Method.invoke (Method.java:566)
> at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
> at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
> at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
> at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
> Caused by: org.apache.maven.plugin.MojoFailureException: There are test failures.
> 
> Please refer to /home/ehtesham/IdeaProjects/checkstyle/target/surefire-reports for the individual test results.
> Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
> at org.apache.maven.plugin.surefire.SurefireHelper.throwException (SurefireHelper.java:289)
> at org.apache.maven.plugin.surefire.SurefireHelper.reportExecution (SurefireHelper.java:161)
> at org.apache.maven.plugin.surefire.SurefirePlugin.handleSummary (SurefirePlugin.java:364)
> at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked (AbstractSurefireMojo.java:1041)
> at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute (AbstractSurefireMojo.java:857)
> at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
> at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
> at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
> at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
> at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
> at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
> at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
> at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
> at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
> at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
> at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
> at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
> at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
> at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
> at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
> at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
> at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
> at java.lang.reflect.Method.invoke (Method.java:566)
> at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
> at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
> at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
> at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
> 
> This is what I get as output .One I understood is due to the wrong commit message but I can't get the other one.

Remove colon after Issue, the first colon in ` Issue:#7609:Update doc for FinalClass\n`. Also why there is "\n", maybe in commit an empty line is there, remove that also if possible.

**Comment by ek08** [comment]
XdocCheck


> open IDE, open class XdocsJavaDocsTest , execute testAllCheckSectionJavaDocs out put will be more understandable.

No error here  but mvn verify -X still gives

[ERROR] Failures: 
[ERROR]   CommitValidationTest.testCommitMessageHasProperStructure:148 Commit 768b33bf448d16b81e66c306ee0c597255fd4367 message: "Issue:#7609:Update doc for FinalClass\n" is invalid
Proper commit message should adhere to the following rules:
    1) Must match one of the following patterns:
        ^Issue #\d+: .*$
        ^Pull #\d+: .*$
        ^(minor|config|infra|doc|spelling|dependency): .*$
    2) It contains only one line of text
    3) Must not end with a period, space, or tab
    4) Commit message should be less than or equal to 200 characters

The rule broken was: 1
[INFO] 
[ERROR] Tests run: 3406, Failures: 1, Errors: 0, Skipped: 0
[INFO] 
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  02:53 min
[INFO] Finished at: 2020-11-18T21:33:41+05:30
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.2:test (default-test) on project checkstyle: There are test failures.
[ERROR] 
[ERROR] Please refer to /home/ehtesham/IdeaProjects/checkstyle/target/surefire-reports for the individual test results.
[ERROR] Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
[ERROR] -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal org.apache.maven.plugins:maven-surefire-plugin:2.22.2:test (default-test) on project checkstyle: There are test failures.

Please refer to /home/ehtesham/IdeaProjects/checkstyle/target/surefire-reports for the individual test results.
Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:215)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.apache.maven.plugin.MojoFailureException: There are test failures.

Please refer to /home/ehtesham/IdeaProjects/checkstyle/target/surefire-reports for the individual test results.
Please refer to dump files (if any exist) [date].dump, [date]-jvmRun[N].dump and [date].dumpstream.
    at org.apache.maven.plugin.surefire.SurefireHelper.throwException (SurefireHelper.java:289)
    at org.apache.maven.plugin.surefire.SurefireHelper.reportExecution (SurefireHelper.java:161)
    at org.apache.maven.plugin.surefire.SurefirePlugin.handleSummary (SurefirePlugin.java:364)
    at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked (AbstractSurefireMojo.java:1041)
    at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute (AbstractSurefireMojo.java:857)
    at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo (DefaultBuildPluginManager.java:137)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:210)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:156)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:148)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:117)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:128)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:305)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:957)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:289)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:193)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke0 (Native Method)
    at jdk.internal.reflect.NativeMethodAccessorImpl.invoke (NativeMethodAccessorImpl.java:62)
    at jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke (DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke (Method.java:566)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
[ERROR] 
[ERROR]

**Comment by ek08** [comment]
I changed the commit message to "Issue #7607: Update doc for Final Class" after I made changes to javadocs still it is showing me this error.

**Comment by AkMo3** [comment]
Can you push these changes to a new branch, I will have a look on them. Then commit them to this branch.

**Comment by ek08** [comment]
I have fixed the identation issues and the format errors.

> Can you push these changes to a new branch, I will have a look on them. Then commit them to this branch.
> Also just now, I cloned your branch, replaced the coding_design.xml with original file as it had a lot of indention issue, then I merged the previous commits into one with right format, and it worked fine. Seems only thing you would have to do is merge previous commits into one with right format. And then force push them.

https://github.com/checkstyle/checkstyle/pull/8994/commits/202246acaf833b850940ff1be5468589fb9fc125
Please have a look into it

File: pom.xml [hunk 2]
```diff
@@ -1399,6 +1399,8 @@
         </dependencies>
       </plugin>
 
+
```
Comment (AkMo3) [inline pom.xml:1402]: Do not edit this file.

File: src/xdocs/config_design.xml [hunk 3]
```diff
@@ -27,121 +26,121 @@
...
-          This check makes sense only for library projects (not application projects)
-          which care of ideal OOP-design to make sure that class works in all cases even misusage.
-          Even in library projects this check most likely will find classes that are designed
-          for extension by somebody. User needs to use suppressions extensively to got a benefit
-          from this check, and keep in suppressions all confirmed/known classes that are deigned
-          for inheritance intentionally to let the check catch only new classes, and bring this to
-          team/user attention.
+                    Nothing wrong could be with founded classes.
+                    This check makes sense only for library projects (not application projects)
+                    which care of ideal OOP-design to make sure that class works in all cases even misusage.
```
Comment (AkMo3) [inline src/xdocs/config_design.xml:34]: This line is getting longer than 100 characters.

File: src/xdocs/config_design.xml [hunk 4]
```diff
@@ -27,121 +26,121 @@
...
-  protected void validate() {
-    super.validate();
-    if (leaves == null) throw new IllegalArgumentException("No leaves!");
-  }
-
-  public void grow() {
-    validate();
-  }
-}
+                    public abstract class Plant {
```
Comment (AkMo3) [inline src/xdocs/config_design.xml:100]: Content inside source should not be indented.

**Review by AkMo3 — CHANGES_REQUESTED** [review]
There are a lot of errors . Go through this test results [Cirrus Check](https://cirrus-ci.com/task/4850777485213696) . You would see the errors, and at which specific line it is occuring.
Also there are right now 7 commits for this PR, squash them into 1.

**Comment by AkMo3** [comment]
What I would advice right now is, just paste the section of code in the appropriate place in the original file( Which has no changes done you) that is to modified for this check and do not alter the whole coding_design.xml file.

**Comment by ek08** [comment]
> What I would advice right now is, just paste the section of code in the appropriate place in the original file( Which has no changes done you) that is to modified for this check and do not alter the whole coding_design.xml file.

Error
The error is due to this stuff in pom.xml I guess but I cannot fix this.I have fixed the indentations.

**Comment by ek08** [comment]
It says
Cannot resolve symbol 'mvn.runtime_classpath'
Cannot resolve symbol 'mvn.project.build.directory'
Cannot resolve symbol 'mvn.project.version'
Cannot resolve symbol 'mvn.project.basedir'

**Comment by AkMo3** [comment]
> 
> 
> It says
> Cannot resolve symbol 'mvn.runtime_classpath'
> Cannot resolve symbol 'mvn.project.build.directory'
> Cannot resolve symbol 'mvn.project.version'
> Cannot resolve symbol 'mvn.project.basedir'

Not sure, but maybe updating directories and sources might solve the issue. In intellij it done by right clicking pom.xml and update select the option from maven tab. Push changes to this branch. I will look into it.

**Comment by romani** [comment]
closed in favor of https://github.com/checkstyle/checkstyle/pull/8996

**Pull request closed without merging**